Assets/Scripts/Augment/BulletModifiers/ZigzagPathModifier.cs

```diff
@@ -0,0 +1,60 @@
+using UnityEngine;
+
+public class ZigzagPathModifier : MonoBehaviour, ProjectileModifier
+{
+    [SerializeField]
+    private float interval = .3f;
+
+    [SerializeField]
+    private float radius = .3f;
+
+    private const string nextOffsetId = "nextZigzagOffset";
+    private const string lastOffsetId = "lastZigzagOffset";
+    private const string lastDistanceId = "lastZigzagDistance";
+
+    public Priority GetPriority() => Priority.EXTENSION;
+
+    private Vector2 RandomOffset(Vector2 lastOffset)
+    {
+        var random = Random.insideUnitCircle * radius;
+        // Restrict to +/-[0.3, 1]
+        random = new Vector2(Mathf.Sign(random.x) * (.7f * Mathf.Abs(random.x) + radius * .3f), Mathf.Sign(random.y) * (.7f * Mathf.Abs(random.y) + radius * .3f));
+        // Prevent the new offset from being on the same half of the unit circle as the previous one.
+        while (Vector2.Dot(random, lastOffset) > 0)
+        {
+            random = Vector2.Perpendicular(random);
+        }
+        return random;
+    }
+
+    public void AddZigzagDisplacement(float distance, ref ProjectileState state)
+    {
+        if (state.distanceTraveled > interval + (float)state.additionalProperties[lastDistanceId])
```

Comment:
Why not use state.distanceTraveled directly? 
The we wouldn't need the lastDistanceId additionalProperty

Reply:
The problem is that we can't tell when we passed the interval with just the distance travelled, since projectiles jump in unpredictable chunks from the modifier's point of view. Could try calculating last step from other variables and use that to determine when we enter a new interval, but that'll make it a more convoluted check

Reply:
> The problem is that we can't tell when we passed the interval with just the distance travelled
But isn't that what you do in the code though? 
Specifically: `state.additionalProperties[lastDistanceId] = state.distanceTraveled;` in the body of the if statement

It seems like the only place it's updated? 🤔

Reply:
It's the distance travelled last time the lightning turned, which is only updated when the if statement triggers.

Reply:
I see now that the value is a timestamp and not a refference, so the implementation is very sensible 👍